app/(chat)/api/chat/route.ts

```diff
@@ -0,0 +1,267 @@
+        - today's date is ${new Date().toLocaleDateString()}.
+        - ask follow up questions to nudge user into the optimal flow
+        - ask for any details you don't know, like name of passenger, etc.'
+        - C and D are aisle seats, A and F are window seats, B and E are middle seats
+        - here's the optimal flow
+          - search for flights
+          - choose flight
+          - select seats
+          - create reservation (ask user whether to proceed with payment or change reservation)
+          - authorize payment (requires user consent, wait for user to finish payment and let you know when done)
+          - display boarding pass (DO NOT display boarding pass without verifying payment)
+        '
+      `,
+    messages: coreMessages,
+    tools: {
+      getWeather: {
+        description: "Get the current weather at a location",
+        parameters: z.object({
+          latitude: z.number().describe("Latitude coordinate"),
+          longitude: z.number().describe("Longitude coordinate"),
+        }),
+        execute: async ({ latitude, longitude }) => {
+          const response = await fetch(
+            `https://api.open-meteo.com/v1/forecast?latitude=${latitude}&longitude=${longitude}&current=temperature_2m&hourly=temperature_2m&daily=sunrise,sunset&timezone=auto`,
+          );
+
+          const weatherData = await response.json();
+          return weatherData;
+        },
+      },
+      displayFlightStatus: {
+        description: "Display the status of a flight",
+        parameters: z.object({
+          flightNumber: z.string().describe("Flight number"),
+          date: z.string().describe("Date of the flight"),
+        }),
+        execute: async ({ flightNumber, date }) => {
+          const flightStatus = await generateSampleFlightStatus({
+            flightNumber,
+            date,
+          });
+
+          return flightStatus;
+        },
+      },
+      searchFlights: {
+        description: "Search for flights based on the given parameters",
+        parameters: z.object({
+          origin: z.string().describe("Origin airport or city"),
+          destination: z.string().describe("Destination airport or city"),
+        }),
+        execute: async ({ origin, destination }) => {
+          const results = await generateSampleFlightSearchResults({
+            origin,
+            destination,
+          });
+
+          return results;
+        },
+      },
+      selectSeats: {
+        description: "Select seats for a flight",
+        parameters: z.object({
+          flightNumber: z.string().describe("Flight number"),
+        }),
+        execute: async ({ flightNumber }) => {
+          const seats = await generateSampleSeatSelection({ flightNumber });
+          return seats;
+        },
+      },
+      createReservation: {
+        description: "Display pending reservation details",
+        parameters: z.object({
```

Comment:
thoughts on defining these schemas once and then reusing across ai/actions.ts and here? could also use the infered types for the component?